src/main/java/org/candlepin/model/RhsmApiCompatCurator.java

```diff
@@ -103,4 +116,421 @@ public List<ConsumerEntitlementCount> getConsumerEntitlementCounts(Collection<St
+                    List<String> possibleUuids = Util.getPossibleUuids(virtUuid);
+                    for (String possible : possibleUuids) {
+                        HypervisorGuestMapping mapping = hypervisorGuestMappings.get(possible.toLowerCase());
+                        if (mapping != null) {
+                            consumerFeed.setHypervisorUuid(mapping.hypervisorUuid())
+                                .setHypervisorName(mapping.hypervisorName())
+                                .setGuestId(mapping.guestId());
+                            break;
+                        }
+                    }
+                }
+            }
+
+            consumerFeed
+                .setFacts(ConsumerFeedFactExtractor.extractRelevantFacts(facts))
+                .setInstalledProducts(installedProductsByConsumer.get(consumerFeed.getId()))
+                .setSyspurposeAddons(addOnsByConsumer.get(consumerFeed.getId()));
+        }
+
+        return consumersFiltered;
+    }
+
+    /**
+     * Retrieves list of {@link ConsumerFeed} objects for a given owner,
+     * filtered optionally by consumer ID, UUID, and last check-in timestamp.
+     * <p>
+     * This method constructs a dynamic HQL query to fetch consumers matching the provided filters,
+     * ordered deterministically by ID and last check-in timestamp (ascending). The query is paginated
+     * using the specified page and perPage parameters.
+     * <p>
+     * Only non-null filter parameters are applied. If no filters are provided, all consumers for the given
+     * owner are returned.
+     *
+     * @param ownerKey     the key of the owner whose consumers are to be fetched; must not be null or empty
+     * @param afterId      (optional) only consumers with an ID lexicographically greater than this value
+     *                     will be included
+     * @param afterUuid    (optional) only consumers with a UUID lexicographically greater than this value
+     *                     will be included
+     * @param afterCheckin (optional) only consumers with a last check-in timestamp after this value will
+     *                     be included
+     * @param page         the page number to retrieve
+     * @param perPage      the number of consumers to return per page
+     * @return a list of {@link ConsumerFeed} matching the given filters and paging criteria
+     */
+    private List<ConsumerFeed> getConsumersFiltered(String ownerKey, String afterId, String afterUuid,
+        OffsetDateTime afterCheckin, int page, int perPage) {
+
+        StringBuilder hql = new StringBuilder(
+            "SELECT new org.candlepin.model.ConsumerFeed(" +
+                "c.id, c.uuid, c.name, c.typeId, c.owner.key, c.lastCheckin, c.serviceLevel, c.role)" +
+            "FROM Consumer c " +
+            "WHERE c.owner.key = :owner");
+
+        if (afterId != null) {
+            hql.append(" AND c.id > :afterId");
+        }
+        if (afterUuid != null) {
```

Comment:
Missing whitespace between the blocks. This applies to several areas throughout this PR.